portal-impl/src/com/liferay/portal/dao/db/BaseDB.java

```diff
@@ -781,6 +781,10 @@ else if (!tablesSQLLowerCase.contains(
 
 	protected abstract String[] getTemplate();
 
+	protected String prepareMaxStringIndexLengthLimitation(String template) {
```

Comment:
@samanzanedo I am confused by this method. It is implemented and used only by OracleDB. Why are you defining it in BaseDB? If this is meant to be an extension point for subclass to overwrite, it at least needs to be used within BaseDB, otherwise subclass won't be able to overwrite anything to change BaseDB's behavior.